### ISSUE TITLE
After 2024-06-19-1857, Prison generation breaks

### ISSUE THREAD

**Issue by MochaModifier** [CleverRaven/Cataclysm-DDA#74686]
### Describe the bug

Prisons (as used in prison starts) completely break in map generation; with walls missing entirely in some cases. The cafeteria spawns without walls adjacent to kitchen, surrounding walls partially or entirely missing.
Frequently, the player character will spawn on the wrong side of the visitation area; and can simply walk out of the prison regardless of whether there are walls - nullifying any challenge behind this challenge start

### Attach save file

n/a

### Steps to reproduce

1. Generate a new world (optional, recommended)
2. Start a new save with the (non-island) Prison start
3. Explore the area and determine what parts are missing.

### Expected behavior

The prison should be entirely enclosed by un-scalable walls, keeping the player inside and forcing them to escape through the front (barring some ability to dig through blocks)

### Screenshots

![24-06-20-1718874354-cataclysm-tiles](https://github.com/CleverRaven/Cataclysm-DDA/assets/38334602/346c301a-c09d-43a3-a80f-54b34a848877)
![24-06-20-1718875123-cataclysm-tiles](https://github.com/CleverRaven/Cataclysm-DDA/assets/38334602/87a71d03-6b60-4649-8cd2-4e6f884ad93e)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.4412 (22H2)
- Game Version: cdda-experimental-2024-06-19-1857 e4840ea [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Bionic Professions [package_bionic_professions],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    SpeedyDex [speedydex],
    Tamable Wildlife [Tamable_Wildlife],
    Slowdown Fungal Growth [no_fungal_growth],
    Extra Mutated Scenarios [extra_mut_scens],
    Translate Complex Dialogue [translate_dialogue],
    Alternative Map Key [alt_map_key]
]

**Comment by Void-Seeker** [comment]
It's likely a duplicate of #74683, should be fixed after #74687 is merged.

**Comment by MochaModifier** [comment]
Closing as duplicate of https://github.com/CleverRaven/Cataclysm-DDA/issues/74683

**Comment by RenechCDDA** [comment]
Closing as duplicate. As the duplicated issue has been resolved, you should update to the version with the fix when the build is available for your system.
